File: src/java/com/cia/servlet/Actualizar.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package com.cia.servlet;

import com.cia.db.Actulizaciones;
import com.cia.db.Conexion;
import com.cia.db.Consultas;
import com.cia.db.Inserciones;
import com.cia.persistencia.CiaInfracciones;
import com.cia.persistencia.CiaPersonas;
import com.cia.utils.Cifrado.Reportes;
import java.io.IOException;
import java.io.PrintWriter;
import java.math.BigDecimal;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;
import java.util.logging.Level;
import java.util.logging.Logger;
import javax.servlet.ServletContext;
import javax.servlet.ServletException;
import javax.servlet.annotation.WebServlet;
import javax.servlet.http.HttpServlet;
import javax.servlet.http.HttpServletRequest;
import javax.servlet.http.HttpServletResponse;
import org.json.JSONArray;
import org.json.JSONObject;

/**
 *
 * @author Desarrollo1
 */
@WebServlet(name = "Actualizar", urlPatterns = {"/actualizar"})
public class Actualizar extends HttpServlet {

    /**
     * Processes requests for both HTTP <code>GET</code> and <code>POST</code>
     * methods.
     *
     * @param request servlet request
     * @param response servlet response
     * @throws ServletException if a servlet-specific error occurs
     * @throws IOException if an I/O error occurs
     */
    protected void processRequest(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException, Exception {
        response.setContentType("text/html;charset=UTF-8");
        try (PrintWriter out = response.getWriter()) {

            if (request.getParameter("action") != null) {
                if (request.getParameter("action").equals("consultar")) {
                    buscarInfraciones(request, response);
                } else if (request.getParameter("action").equals("update")) {
                     response.setContentType("aplication/json");
                    String data = request.getParameter("tipo_curso");
                    JSONObject jSONObject = new JSONObject(data);
                    Conexion con = new Conexion();
                    Actulizaciones actulizaciones = new Actulizaciones();
                    Inserciones in = new Inserciones();
                    con.conectar();
                    String numeroDocumento = jSONObject.get("numero_documento").toString();
                    //String horario = jSONObject.get("horario").toString();
                    String tipoDocumento = jSONObject.get("tipo_documento").toString();
                    String nombre_persona = jSONObject.get("nombres_persona").toString();
                    String apellido_persona = jSONObject.get("apellido_persona").toString();
                    String numero_infracion = jSONObject.get("numero_infracion").toString();
                    String numero_factura = jSONObject.get("numero_factura").toString();
                    String fecha_factura = jSONObject.get("fecha_factura").toString();
                    String valor_curso = jSONObject.get("valor_curso").toString();
                    String numero_comparendo = jSONObject.get("numero_comparendo").toString();
                    String fecha_comparendo = jSONObject.get("fecha_comparendo").toString();
                    boolean personaupdate =actulizaciones.updatePersona(con.getCon(), nombre_persona, apellido_persona, tipoDocumento, numeroDocumento);
                    boolean infracionesupdate = actulizaciones.updateInfraciones(con.getCon(), numero_factura, fecha_factura, valor_curso, numero_infracion, numero_comparendo, fecha_comparendo);
                    if (infracionesupdate && personaupdate) {
                        out.print("datos actualizados correctamente");
                    }else{
                         out.print("error contacte al aministrador del sistema");
                    }
                }

                } else {
                    request.getRequestDispatcher("paginas/actualizarInf.jsp").forward(request, response);
                }
            }
        }

        // <editor-fold defaultstate="collapsed" desc="HttpServlet methods. Click on the + sign on the left to edit the code.">
        /**
         * Handles the HTTP <code>GET</code> method.
         *
         * @param request servlet request
         * @param response servlet response
         * @throws ServletException if a servlet-specific error occurs
         * @throws IOException if an I/O error occurs
         */
        @Override
        protected void doGet
        (HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
            try {
                processRequest(request, response);
            } catch (Exception ex) {
                Logger.getLogger(Actualizar.class.getName()).log(Level.SEVERE, null, ex);
            }
        }

    

    protected void buscarInfraciones(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        try {
            response.setContentType("aplication/json");
            Conexion conexion = new Conexion();
            conexion.conectar();
            Consultas consultas = new Consultas();
            String numero = request.getParameter("numero_documento");
            PrintWriter out = response.getWriter();
            CiaPersonas ciaPersonas = consultas.ConsultaPersonaPorDocumento(conexion.getCon(), numero, BigDecimal.ONE);
            if (ciaPersonas != null) {
                List<CiaInfracciones> listInfraciones = consultas.infraccionesByPersona(conexion.getCon(), ciaPersonas.getPerId().intValue());
                if (listInfraciones.size() > 0) {
                    JSONArray jsonArray = new JSONArray();
                    for (CiaInfracciones hashMap : listInfraciones) {
                        JSONObject jSONObject = new JSONObject();
                        jSONObject.put("drc_id", hashMap.getInfNumero());
                        jSONObject.put("per_nombres", ciaPersonas.getPerNombres());
                        jSONObject.put("per_apellidos", ciaPersonas.getPerApellidos());
                        jSONObject.put("per_documetos", ciaPersonas.getPerDocumento());
                        jSONObject.put("per_tipo_documento", ciaPersonas.getPerTpDocumento().toString());
                        jSONObject.put("numero_factura", hashMap.getInfFactura());
                        jSONObject.put("fecha_factura", hashMap.getInfFacFecha());
                        if (hashMap.getInfvalorCurso() != null) {
                            jSONObject.put("valor_curso", hashMap.getInfvalorCurso().toString());
                        }

                        jSONObject.put("codigo_infracion", hashMap.getInfCodigo().toString());
                        jSONObject.put("fecha_comparendo", hashMap.getInfFecha());
                        jsonArray.put(jSONObject);

                    }

                    out.print(jsonArray);
                }
            }

        } catch (Exception ex) {
            Logger.getLogger(Actualizar.class.getName()).log(Level.SEVERE, null, ex);
        }
    }

    /**
     * Handles the HTTP <code>POST</code> method.
     *
     * @param request servlet request
     * @param response servlet response
     * @throws ServletException if a servlet-specific error occurs
     * @throws IOException if an I/O error occurs
     */
    @Override
    protected void doPost(HttpServletRequest request, HttpServletResponse response)
            throws ServletException, IOException {
        try {
            processRequest(request, response);
        } catch (Exception ex) {
            Logger.getLogger(Actualizar.class.getName()).log(Level.SEVERE, null, ex);
        }
    }

    /**
     * Returns a short description of the servlet.
     *
     * @return a String containing servlet description
     */
    @Override
    public String getServletInfo() {
        return "Short description";
    }// </editor-fold>

}

File: web/paginas/js/actualizar.js
var app = new Vue({
    el: "#app",
    data: {
        items: [],
        selected: [],
        renderDatos: false,
        detalle_curso_persona: {
            horario: '',
            tipo_documento: '',
            numero_documento: '',
            nombres_persona: '',
            apellido_persona: '',
            numero_infracion: '',
            numero_factura: '',
            fecha_factura: '',
            valor_curso: '',
            numero_comparendo: '',
            fecha_comparendo: ''

        },
        detalle_curso_persona_current: {
            horario: '',
            tipo_documento: '',
            numero_documento: '',
            nombres_persona: '',
            apellido_persona: '',
            numero_infracion: '',
            numero_factura: '',
            fecha_factura: '',
            valor_curso: '',
            numero_comparendo: '',
            fecha_comparendo: '',

        }


    },
    methods: {
        select: function () {
            this.selected = [];
            if (!this.selectAll) {
                for (let i in this.items) {
                    this.selected.push(this.items[i]);
                }
            }
        },
        guardar: function () {
            var tipo_curso = document.getElementById("cmbTipoDocumento").value;
            var numero_documento = document.getElementById("txtDocumento").value;
            axios.get("<%= request.getContextPath()%>/actualizar", {params: {tipo_curso: tipo_curso, numero_documento: numero_documento, action: "consultar"}}).
                    then(response => {
                        if (response.data.length > 0) {
                            this.items = [];
                            for (var i = 0; response.data.length; i++) {
                                this.detalle_curso_persona = {};
                                this.detalle_curso_persona.tipo_documento = response.data[i].per_tipo_documento;
                                this.detalle_curso_persona.numero_documento = response.data[i].per_documetos;
                                this.detalle_curso_persona.numero_infracion = response.data[i].codigo_infracion;
                                this.detalle_curso_persona.nombres_persona = response.data[i].per_nombres;
                                this.detalle_curso_persona.apellido_persona = response.data[i].per_apellidos;
                                this.detalle_curso_persona.fecha_comparendo = response.data[i].fecha_comparendo;
                                this.detalle_curso_persona.numero_factura = response.data[i].numero_factura;
                                this.detalle_curso_persona.fecha_factura = response.data[i].fecha_factura;
                                this.detalle_curso_persona.valor_curso = response.data[i].valor_curso;
                                this.detalle_curso_persona.numero_comparendo = response.data[i].drc_id;
                                this.items.push(this.detalle_curso_persona);
                            }

                        } else {
                            this.items = [];
                            alert("el tipo y el documento no fueron encontrado en la base de datos");
                        }

                    }).catch(error => {
                console.log(error);
            });
        },
        guardarAsistencias: function () {

            var data = this.selected;
            if (data.length > 0) {
                axios.get("<%= request.getContextPath()%>/asistencias", {params: {data: JSON.stringify(data), action: "procesarPersona"}}).
                        then(response => {

                            this.report();
                        }).catch(error => {
                });
            } else {
                alert("debe elegir al menos a una persona");
            }

        },
        editarDatos: function () {
            axios.get("<%= request.getContextPath()%>/actualizar", {params: {tipo_curso: this.detalle_curso_persona_current, action: "update"}}).
                    then(response => {
                        if (response.data.length > 0) {
                            this.items = [];
                            for (var i = 0; response.data.length; i++) {
                                this.detalle_curso_persona = {};
                                this.detalle_curso_persona.numero_documento = response.data[i].per_documetos;
                                this.detalle_curso_persona.tipo_documento = response.data[i].per_tipo_documento;
                                this.detalle_curso_persona.numero_infracion = response.data[i].drc_id;
                                this.detalle_curso_persona.nombres_persona = response.data[i].per_nombres;
                                this.detalle_curso_persona.apellido_persona = response.data[i].per_apellidos;
                                this.detalle_curso_persona.numero_factura = response.data[i].numero_factura;
                                this.detalle_curso_persona.fecha_factura = response.data[i].fecha_factura;
                                this.detalle_curso_persona.valor_curso = response.data[i].valor_curso;
                                this.detalle_curso_persona.numero_comparendo = response.data[i].valor_curso;
                                this.items.push(this.detalle_curso_persona);
                            }

                        } else {
                            this.items = [];
                            alert("el tipo y el documento no fueron encontrado en la base de datos");
                        }

                    }).catch(error => {
                console.log(error);
            });
        },
        test: function (i) {
            this.detalle_curso_persona_current = i;
            this.items = [];
            this.renderDatos = true;

        }
    }
})
